content/tutorials/peer-to-peer-vue.textile

```diff
@@ -0,0 +1,671 @@
+
+Here, we're taking a phrase, and sending it as a series of messages word-by-word, with a 500ms pause in between each word. We're using @async / await@ to pause our code, just be *resolving a promise* every 500ms.
+
+Now we should update the client to listen for this message. Add a @case@ statement for the @word@ message to it's @onReceiveMessage@ handler.
+
+```[javascript]
+    onReceiveMessage(message) {
+      if (message.serverState) {
+        this.serverState = message.serverState; 
+      }
+
+      switch(message.kind) {
+        case "connection-acknowledged": 
+          this.state.status = "acknowledged"; 
+          break;
+        case "word":
+          this.state.receivedWords += " " + message.word;
+          break;
+        default: () => { };
+      }
+    }
+```
+
+When we receive a @word@ message, it is added to a string in our state object. Because of this, we also need to update the @this.state@ object in the @P2PClient@ constructor to start us off with an empty @receivedWords@ property.
+
+```[javascript]
+this.state = {
+  status: "disconnected",
+  receivedWords: ""
+  };
+```
+
+Next, we're going to make sure to display this in the UI of every attached client by binding the state property into the markup in @index.html@
+
+```html
+  <section v-else class="game-info">
+    <h2>UniqueId: {{ uniqueId }}</h2>    
+    <button v-if="iAmHost" v-on:click="sendWordsAsHost" class="form-button">Stream words to client</button>     
+
+    <h3>Active players: {{ transmittedServerState?.players?.length }}</h3>
+    <ul class="players">
+      <li class="player" v-for="user in transmittedServerState?.players">                
+        <span>{{ user.friendlyName }}</span>
+      </li>
+    </ul>
+
+    <div>
+      {{ this.p2pClient.state.receivedWords }}
+    </div>
+  </section>
+</div>
+```
+
+We're doing two things here:
+
+1. Adding a @sendWordsAsHost@ onclick to a new @html button@
+2. Binding @{{ this.p2pClient.state.receivedWords }}@ - the string of words that the client has seen.
+
+We need to create a function to handle the @Stream words to client@ click, so add a new method to the @Vue methods@ property in @index.js@
+
+```[javascript]
+sendWordsAsHost: async function(evt) {
+  await this.p2pServer.sendWordsAcrossMultipleMessages();
+}
+```
+
+This calls the function that added to our @P2PServer@ and the button only displays to the host because of the @v-if@ directive bound to a new @computed property@ @iAmHost@ that verifies that a @p2pserver@ is active.
+
+We'll add @iAmHost@ to the @computed@ property in @index.js@.
+
+```[javascript]
+computed: {
+  state: function() { ... },
+  transmittedServerState: function() { ... },
+  joinedOrHosting: function () { ... },
+  iAmHost: function() { return this.p2pServer != null; },
+}
+```
+
+So now, if you start a hosting a session, and click @Stream words to client@ you'll see a stream of words appear one by one in your UI.
```

Comment:
`So now, if you start a hosting a session` => `So now, if you start hosting a session`